src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -241,15 +241,25 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
         val traitRef = impl.traitRef ?: return
         val trait = traitRef.resolveToTrait ?: return
         val traitName = trait.name ?: return
+
+        fun mayDangleOnTypeOrLifetimeParameters(impl: RsImplItem): Boolean {
+            return impl.typeParameters.any() { it.queryAttributes.hasAtomAttribute("may_dangle") } ||
```

Comment:
I am not sure if this added queryAttributes could have a performance impact.

Reply:
Should be fine I think.